charon/src/translate_functions_to_ullbc.rs

```diff
@@ -1888,7 +1892,13 @@ impl<'tcx, 'ctx> TransCtx<'tcx, 'ctx> {
         let generics = bt_ctx.get_generics();
         let preds = bt_ctx.get_predicates();
 
-        let body = if rust_id.is_local() && is_transparent {
+        // Check if the type is opaque or foreign
+        let has_local_body = rust_id.as_local().is_some_and(|local_def_id| {
+            let node = bt_ctx.t_ctx.tcx.hir().find_by_def_id(local_def_id);
+            !matches!(node, Some(HirNode::ForeignItem(_)))
+        });
```

Comment:
I think it's worth putting this check in an auxiliary helper to factor it out and because in the future we might want to tweak it (I have a feature in mind).